File: 8th-May-Binary_Modulo/php/solution.php
/*
This Code is written by Bhaskar
*/

class Solution {
    public function modulo($s, $m) {
        $k = 0;

        // Convert the binary string to an integer
        for ($i = 0; $i < strlen($s); $i++) {
            $k = (($k << 1) % $m + ($s[$i] - '0') % $m) % $m;
        }

        return $k;
    }
}

File: 8th-May-Binary_Modulo/c++/solution.cpp
/*
This Code is written by Bhaskar
*/

class Solution {
public:
    int modulo(string s, int m) {
        int k = 0;

        // Convert the binary string to an integer
        for (int i = 0; i < s.length(); i++) {
            k = ((k << 1) % m + (s[i] - '0') % m) % m;
        }

        return k;
    }
};

